cogs/polling.py

```diff
@@ -0,0 +1,56 @@
+import discord
+import aiohttp
+import asyncio
+from urllib.request import urlopen
+import json, time
+
+from discord.ext import commands
+from .utils import checks
+
+
+class CommitTracker:
+    def __init__(self, bot):
+        self.bot = bot
+        self.base_commit = bot.config['basecommit']
+        self.wait_time = 120
+
+    async def get_latest_commit(self, owner, repo):
+        url = 'https://api.github.com/repos/{owner}/{repo}/commits?per_page=1'.format(owner=owner, repo=repo)
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url) as response:
+                return await response.text()
+
+    async def trackCommits(self):
+        await self.bot.wait_until_ready()
+        commit = self.bot.config['basecommit']
+        while True:
+            oldcommit=commit
+            owner = 'kwsch'
+            repo = 'PKHeX'
+            data = await self.get_latest_commit(owner, repo) 
+            try:
+                commitdata = json.loads(data)[0]
+            except KeyError:
+                print(json.loads(data))
+            commit = commitdata['sha']
+            if commit != oldcommit:
+                oldcommit = commit
+                self.bot.config['basecommit'] = commit
+                with open("config.json", 'r+') as conf:
+                    newconfig = json.load(conf)
+                    newconfig['basecommit'] = commit
+                    conf.seek(0)
+                    conf.truncate()
+                    json.dump(newconfig, conf, indent=4)
+                embed = discord.Embed(color=7506394)
+                embed.title = "[{repo}:master] 1 new commit".format(repo=repo)
```

Comment:
This assumes that only one commit will be made within 2 minutes. A reasonable assumption, I think, just something to take note of in future.

Reply:
[this](https://github.com/chenzw95/porygon/blob/3808fbf5e42803b60c1fe6ae33b03ff910a563a5/cogs/polling.py#L18) line will only pull one result and so there will be only one commit. I was wondering if there would be a better implementation for this, but didn't end up thinking of one